imu_sensor_broadcaster/src/imu_sensor_broadcaster.cpp

```diff
@@ -43,7 +44,19 @@ controller_interface::CallbackReturn IMUSensorBroadcaster::on_init()
 controller_interface::CallbackReturn IMUSensorBroadcaster::on_configure(
   const rclcpp_lifecycle::State & /*previous_state*/)
 {
-  params_ = param_listener_->get_params();
+  try
+  {
+    params_ = param_listener_->get_params();
+    r_ =
+      quat_from_euler(params_.calibration.roll, params_.calibration.pitch, params_.calibration.yaw);
```

Comment:
```suggestion
    r_ =
      quat_from_euler(params_.calibration.roll, params_.calibration.pitch, params_.calibration.yaw);
    r_.normalize();
```

I made it a habit to normalize the quaternions